### PR TITLE
Comitting changes for Verify Push, Verify Softtoken, Verify Registration, TLS.

### PR DESCRIPTION
**Pull request by Manmohan-TechM**

### CODE REVIEW

File: src/com/telesign/util/TeleSignRequest.java [hunk 1]
```diff
@@ -497,4 +502,26 @@ private String encode(String data, String key)  throws java.security.SignatureEx
 
 		return result;
 	}
+	
+	/**
+	 * Set the TLS protocol to TLSv1.2
+	 */
+	private void setTLSProtocol() {
```
Comment (hmorales) [inline src/com/telesign/util/TeleSignRequest.java]: I thought the idea was to let the user select from supported versions 1.1 and 1.2 and not force them to 1.2?

**Comment by Manmohan-TechM** [comment]
Users who are overriding TLS versions via jvm options will still be able to use TLSv1.1 . As jvm options take precedence over connection specific parameters.

Also please note that any person who has jre7 installed on his system supports both TLSv1.1 & TLSv1.2.

Jre8 uses TLSv1.2 as default.

File: src/com/telesign/util/TeleSignRequest.java (hunk 1 above)
Comment (Manmohan-TechM) [inline src/com/telesign/util/TeleSignRequest.java]: Users who are overriding TLS versions via jvm options will still be able to use TLSv1.1 . As jvm options take precedence over connection specific parameters.

Also please note that any person who has jre7 installed on his system supports both TLSv1.1 & TLSv1.2.

Jre8 uses TLSv1.2 as default.

File: src/com/telesign/verify/Verify.java [hunk 2]
```diff
@@ -222,13 +241,15 @@ public VerifyResponse call(String phone_number, String language, String originat
...
-	 * 								   Set it to null if not sending originating ip.
-	 * @param session_id	[Optional] Your end users session id. Set it to "null" if not sending session id.
+	 * @param originating_ip 		[Optional]  Your end users IP Address. This value must be in the format defined by IETF in the 
+	 * 								   		    Internet-Draft document titled Textual Representation of IPv4 and IPv6 Addresses. Ex: originating_ip=192.168.123.456.
+	 * 								 		    Set it to null if not sending originating ip.
+	 * @param session_id			[Optional]  Your end users session id. Set it to "null" if not sending session id.
+	 * @param call_forward_action 	[Optional]	A string containing call forward action
 	 * @return A {@link com.telesign.verify.response.VerifyResponse} object, which contains the JSON-formatted response body from the TeleSign server.
 	 */
-	public VerifyResponse call(String phone_number , String language, String verify_code, String verify_method, int extension_type, String extension_template, boolean redial, String originating_ip, String session_id) {
```
Comment (hmorales) [inline src/com/telesign/verify/Verify.java]: We can't remove existing interfaces, please overload. We already had a complaint from a previous commit I missed where you broke the API for a customer. Would be good it you could address that as well.

https://github.com/TeleSign/java_telesign/commit/1e13590b62ef9abffad9a47e2e4cd38219383ad0#diff-c4f0d7816b294665a021081861dc242c

File: src/com/telesign/verify/Verify.java [hunk 3]
```diff
@@ -316,58 +410,58 @@ public VerifyResponse status(String resource_id, String verify_code) {
...
-	 * @param originating_ip [Optional] Your end users IP Address. This value must be in the format defined by IETF in the 
-	 * 								   Internet-Draft document titled Textual Representation of IPv4 and IPv6 Addresses. Ex: originating_ip=192.168.123.456.
-	 * 								   Set it to null if not sending originating ip.
-	 * @param session_id	[Optional] Your end users session id. Set it to "null" if not sending session id.
-	 * @return A {@link com.telesign.verify.response.VerifyResponse} object, which contains the JSON-formatted response body from the TeleSign server.
+	 * @param phone_number [Required] Your end user�s phone number, including the country code.
+	 * @param bundle_id    [Required]
+	 * @return  A {@link com.telesign.verify.response.VerifyResponse} object, which contains the JSON-formatted response body from the TeleSign server.
 	 */
-	public VerifyResponse status(String resource_id, String originating_ip, String session_id) {
```
Comment (hmorales) [inline src/com/telesign/verify/Verify.java]: Why are you removing this method?

**Pull request merged**